src/operators/Cylinder/PyCylinderAttributes.C

```diff
@@ -316,8 +316,15 @@ PyCylinderAttributes_setattr(PyObject *self, char *name, PyObject *args)
         Py_DECREF(obj);
 
     Py_DECREF(tuple);
-    if( obj == NULL)
-        PyErr_Format(PyExc_RuntimeError, "Unable to set unknown attribute: '%s'", name);
+    if (obj == PyExc_NameError)
+        obj = PyErr_Format(obj, "Unknown attribute: '%s'", name);
+    else if (obj == PyExc_TypeError)
+        obj = PyErr_Format(obj, "Bad type for attribute: '%s'", name);
+    else if (obj == PyExc_IndexError)
+        obj = PyErr_Format(obj, "Bad index for attribute: '%s'", name);
+    else if (obj == PyExc_ValueError)
```

Comment:
ValueError is returned for invalid length, but doesn't give any indication that the length is the reason for the error.

Why have an if-statement for IndexError that is never set for this class?

Reply:
> ValueError is returned for invalid length, but doesn't give any indication that the length is the reason for the error.

We could use `IndexError` for this instead. That might make some sense. The message could read `"Bad size or index for attribute '%s'"`. An earlier incantation of this did just that.

Most of the `Set<FieldName>()` logic in our Python CLI objects first attempts a multi-value assignment *without* expecting a sequence type (list, tuple, etc.) as in `a,b,c=1,2,3`. If that fails, it tries obtaining a tuple, checks that the size of the tuple matches what the field-member expects and then tries assigning the members of the tuple using some common conversions.

> Why have an if-statement for IndexError that is never set for this class?

See above

Reply:
> ValueError is returned for invalid length, but doesn't give any indication that the length is the reason for the error.

In re-examining the proposed changes I made...`ValueError` is being used currently *everywhere* for cases of incorrect length. So, as things stand now, we can adjust the error message from `Bad value for attribute` to maybe `Bad value size for attribute`

Reply:
I adjusted logic to address this.